ui/src/components/statistics-card.tsx

```diff
@@ -44,8 +46,11 @@ export const StatisticsCard = ({
         </CardTitle>
       </CardHeader>
       <CardContent className='text-sm'>
-        <div className='flex items-center justify-between'>
-          <div className='text-2xl font-bold'>{value ? value : '-'}</div>
+        <div className='flex flex-col'>
+          <div className='text-2xl font-bold'>
+            {value ? value === 'loading' ? <LoadingIndicator /> : value : '-'}
+          </div>
```

Comment:
This should not be based on matching a magic string like "loading". The simplest solution could be to type value as `React.ReactNode`.